transformer/diff.go

```diff
@@ -2,76 +2,112 @@ package transformer
 
 import (
 	"fmt"
+	"regexp"
+	"strconv"
 	"strings"
 )
 
 type DiffTransformer struct {
-	rawDiff   string
-	fileDiffs []string
-	segments  []string
+	rawDiff         string
+	numberedRawDiff string
+	fileDiffs       map[string]string
+	fileSegments    map[string][]string
 }
```

Comment:
This could use some comments to explain what is being stored in these maps.
DEBUG INFO
[File] transformer/diff.go

[Side] LEFT

[Line] 12

[Raw Comment] [+ Line 12-13] This could use some comments to explain what is being stored in these maps.
[
[Original Code]  a/transformer/diff.go b/transformer/diff.go
index fbaff5a..37bb5dd 100644
 
 import (
 	"fmt"
5 +	"regexp"
6 +	"strconv"
 	"strings"
 )
 
 type DiffTransformer struct {
9 -	rawDiff   string
10 -	fileDiffs []string
11 -	segments  []string
11 +	rawDiff         string
12 +	numberedRawDiff string
13 +	fileDiffs       map[string]string
14 +	fileSegments    map[string][]string
 }
 
 const segmentLength = 4000
 
 func (dt *DiffTransformer) Transform(rawDiff string) {
 	dt.rawDiff = rawDiff
18 -
19 -	// Split diff into files
21 +	dt.numberRawDiff()
 	dt.splitIntoFiles()
 	dt.generateSegments()
 }
 
24 -func (dt *DiffTransformer) GetLastSegment() string {
25 -	return dt.segments[len(dt.segments)-1]
26 -}
27 -
28 -func (dt *DiffTransformer) GetSegments() []string {
29 -	return dt.segments
26 +func (dt *DiffTransformer) GetFileSegments() map[string][]string {
27 +	return dt.fileSegments
 }
 
 func (dt *DiffTransformer) splitIntoFiles() {
33 -	dt.fileDiffs = strings.Split(dt.rawDiff, "